tutorials/polkadot-sdk/system-chains/asset-hub/batch-teleport-assets.md

```diff
@@ -0,0 +1,218 @@
+└── README.md
+```
+
+## Set Up the Project
+
+Create a new Bun project:
+
+```bash
+mkdir batch-teleport-tutorial
+cd batch-teleport-tutorial
+bun init -y
+```
+
+## Install Dependencies
+
+Install the ParaSpell SDK:
+
+```bash
+bun add @paraspell/sdk@{{dependencies.javascript_packages.paraspell_sdk.version}}
+```
+
+## Create the Implementation
+
+Create an `index.ts` file with the complete batch teleport implementation. For testing purposes, you will be using [Paseo Asset Hub](/develop/networks/#paseo){target=\_blank} (a Polkadot community-driven TestNet).
+
+```typescript title="index.ts"
+--8<-- 'code/tutorials/polkadot-sdk/system-chains/asset-hub/index.ts'
+```
+
+!!!note "Understanding Batch Modes"
+    The ParaSpell SDK supports two batch modes:
+
+    - BatchMode.BATCH_ALL
+        - **Behavior**: All operations must succeed, or the entire batch fails
+        - **Use Case**: When all transfers are critical and interdependent
+        - **Gas Efficiency**: More efficient as it's a single atomic operation
+
+    - BatchMode.BATCH
+        - **Behavior**: Individual operations can fail independently
+        - **Use Case**: When some transfers can fail without affecting others
+        - **Resilience**: More resilient but with slightly higher gas overhead
+
+Update your `package.json` to include the necessary scripts:
+
+```json title="package.json"
+{
+  "name": "batch-teleport-tutorial",
+  "version": "1.0.0",
+  "description": "Tutorial for creating batch teleport transactions with ParaSpell SDK using Bun",
+  "main": "index.ts",
+  "type": "module",
+  "scripts": {
+    "start": "bun run index.ts",
+    "dev": "bun run --watch index.ts" // --watch automatically restarts the script when files change
+ },
+  },
+  "dependencies": {
+    "@paraspell/sdk": "^{{dependencies.javascript_packages.paraspell_sdk.version}}"
+  }
+}
+```
+
+## Configure Destination Addresses
+
+For testing purposes, you can also modify the destination addresses in the batch operations:
+
+- `'15whavTNSyceP8SL3Z1JukFcUPzmeR26RxKXkfQiPhsykg7s'`
+- `'14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3'`
+
+## Run Your Batch Teleport
+
+Execute the batch teleport from your project directory:
+
+```bash
+bun run dev
+```
+
+You should see output similar to:
+
+--8<-- 'code/tutorials/polkadot-sdk/system-chains/asset-hub/papi-console-output.html'
+
+The encoded data is formatted into a PAPI console link for easy submission through a web interface.
+
+Make sure your Polkadot wallet is funded with some test tokens by using the [Polkadot faucet](https://faucet.polkadot.io/?parachain=1000){target=\_blank}  and submit the transaction.
+
+## Submitting the Transaction
+
+After running the script:
+
+1. **Copy the generated link** from the console output
+2. **Open the link** in your web browser
+3. **Connect your wallet** (such as Polkadot.js extension)
+4. **Review the transaction details** in the PAPI console
+5. **Submit the transaction** and confirm with your wallet
```

Comment:
```suggestion
1. Copy the generated link from the console output.
2. Open the link in your web browser.
3. Connect your wallet (such as Polkadot.js extension).
4. Review the transaction details in the PAPI console.
5. Submit the transaction and confirm with your wallet.
```

This doesn't really follow the `**Title**: Description.` format, so there's no need to put part of the sentence in bold, can be left as is.